src/Authenticator/CookieAuthenticator.php

```diff
@@ -154,7 +157,21 @@ protected function _createPlainToken($identity): string
         $usernameField = $this->getConfig('fields.username');
         $passwordField = $this->getConfig('fields.password');
 
-        return $identity[$usernameField] . $identity[$passwordField];
+        $salt = $this->getConfig('salt', '');
+        if ($salt === true) {
+            $salt = Security::getSalt();
+        } elseif ($salt === false) {
+            $salt = '';
+        } elseif (!is_string($salt) || $salt === '') {
+            throw new InvalidArgumentException('Salt must be a non-empty string.');
+        }
+
+        $value = $identity[$usernameField] . $identity[$passwordField];
+
+        $hmac = hash_hmac('sha1', $value, $salt);
```

Comment:
Shouldn't generating and appending the hmac be skipped if `$salt` is empty string to preserve backwards compatibility?

Reply:
Yeah, makes sense. I changed it.